lib/common/serializers/helpers.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+module Common
+  module Serializer
+    def date_attr(*names, override_name: nil)
+      name = override_name || names.last
+      define_method(name) do
+        date = object_data.dig(*names)
+        return unless date
+        Date.strptime(date, '%m/%d/%Y')
+      end
+    end
+
+    def yes_no_attr(*names, override_name: nil)
+      name = override_name || names.last
+      define_method(name) do
+        s = object_data.dig(*names)
+        return unless s
+        case s.downcase
+        when 'yes' then true
+        when 'no' then false
+        else
+          Rails.logger.error "Expected key '#{keys}' to be Yes/No. Got '#{s}'."
+          nil
+        end
+      end
```

Comment:
I can't help to think that it would have been better to use with and maybe define these as custom coercion types maybe. Similar to UTCTime.